fluss-server/src/main/java/org/apache/fluss/server/metrics/group/TableMetricGroup.java

```diff
@@ -222,6 +242,42 @@ public Counter failedPrefixLookupRequests() {
         }
     }
 
+    // ------------------------------------------------------------------------
+    //  user groups
+    // ------------------------------------------------------------------------
+    private UserMetricGroup getOrCreateUserMetricGroup(String principalName) {
```

Comment:
IIUC, we only need to collect user-level IO metrics, rather than (user * table)-level. Then I think we can extract this out of `TableMetricGroup` into a global `UserMetrics` (which maintains a map of user to `UserMetricGroup`) in `ReplicaManager`. This will also help us to implement user quota in the future, which has similar implementation in kafka.